docs(material/theming): revise customizing-component-styles (#22466)
angular/components: guides/customizing-component-styles.md

# Customizing Angular Material component styles

Angular Material supports customizing component styles via Sass API as described in the [theming
guide][]. This document provides guidance on defining custom CSS rules that directly style
Angular Material components.

[theming guide]: https://material.angular.io/guide/theming

## Targeting custom styles

### Component host elements

For any Angular Material component, you can safely define custom CSS for a component's host element
that affect the positioning or layout of that component, such as `margin`, `position`, `top`,

> `left`, `transform`, and `z-index`. You should apply such styles upon be defining a custom CSS

Reviewer: ```suggestion
`left`, `transform`, and `z-index`. You should apply such styles by defining a custom CSS
```

in unexpected ways as the internal styles change between releases.

### Internal component elements

Avoid any custom styles or overrides on internal elements within a Angular Material components.
The DOM structure and CSS classes applied for each component may change at any time, causing custom
styles to break.

## Applying styles to Angular Material components

While Angular Material does not support defining custom styles or CSS overrides on components'
internal elements, you might choose to do this anyway. There are three points to consider while
customizing styles for Angular Material components: view encapsulation, CSS specificity, and
rendering location.

### View encapsulation

By default, Angular scopes component styles to exclusively affect that component's view. This means
that the styles you author affect only the elements directly within your component template.
Encapsulated styles do *not* affect elements that are children of other components within your
template. You can read more about view encapsulation in the
[Angular documentation](https://angular.io/guide/component-styles#view-encapsulation). You may

> also wish to take a review

Reviewer: ```suggestion
also wish to review
```

> [_The State of CSS in Angular_](https://blog.angular.io/the-state-of-css-in-angular-4a52d4bd2700)

Reviewer: We're probably due for an update on _The State of CSS in Angular_ with these new changes, this article is a bit dated (both date and content).

Author: Yeah, we should probably update our guidance once we drop IE11 support and we can throw our weight behind css variables (and maybe constructible stylesheets will be a thing by then, too)
